app/tools/circuits/all-circuits-map.php

```diff
@@ -108,9 +148,30 @@
                         $html[] = "  strokeWeight: 3";
                         $html[] = "});";
                     }
+                    */
+                }
+                //Add all circuits to map with type information
+                foreach ($circuits as $circuit) {
+                  //If map_spepcifc is set and its in the array OR it isn't set, map all
+                  if((isset($_GET['map_specific']) && in_array($circuit->id,$circuits_to_map)) || (!isset($_GET['map_specific']))){
+                    $html[] = "path = [[".$all_locations[$circuit->location1]->lat.", ".$all_locations[$circuit->location1]->long."], [".$all_locations[$circuit->location2]->lat.", ".$all_locations[$circuit->location2]->long."]]";
+                    $html[] = "map.drawPolyline({";
+                    $html[] = "  path: path,";
+                    $html[] = "  strokeColor: '".$type_hash[$circuit->type]->ctcolor."',";
+                    if($type_hash[$circuit->type]->ctpattern == "Dotted") { $html[] = "  strokeOpacity: 0,"; }
+                    else{ $html[] = "  strokeOpacity: 0.6,"; }
+                    if($type_hash[$circuit->type]->ctpattern == "Dotted") {
+                      $html[] = " icons:[{
+                        icon: lineSymbol,
+                        offset: '0',
+                        repeat: '20px'
+                        }],"; }
+                    $html[] = "  strokeWeight: 3";
+                    $html[] = "});";
+
+                    print implode("\n", $html);
```

Comment:
I forgot to mention this in my recent comment on the PR... But I'm slightly concerned by this move of the "print implode" line to be inside the if-else block. -- It may be perfectly fine (I haven't fully reviewed the code paths myself), but it should be carefully checked for introducing any failed code paths when compared to the original line which was outside of the block.